### ISSUE TITLE
Add map-based user-friendly UI to define geo-fences

### ISSUE THREAD

**Issue by bazsodombiandras** [domoticz/domoticz-android#396]
The current UI for defining geo-fences is confusing (Get + Manual buttons).
Add a map-based user interface like shown on the attached screenshot (like IFTTT has).
![mapbasedgeofenceui](https://cloud.githubusercontent.com/assets/2814023/22058595/2dd2d524-dd72-11e6-8853-6396c55ba35a.png)

**Comment by galadril** [comment]
Making progress on a new location picker implementation

**So far**
![geo](https://cloud.githubusercontent.com/assets/14561640/22419510/21bd7e72-e6de-11e6-9426-a0b2490a009c.png)

**Comment by bazsodombiandras** [comment]
Looks very nice!The work that you put into this app can really be felt (it keeps getting better and better) and it brings joy to many home automation enthusiasts!Keep it up!

**Comment by galadril** [comment]
Thanks !

Please test on the next beta.. ill close this case for now 👍

**Comment by galadril** [comment]
You can find information on how to become beta tester here:
http://hnogames.nl/domoticz.html

(Sign up for the beta tester channel via google+ communities)

**Comment by bazsodombiandras** [comment]
Hi!
I started testing the new location picker. In general it looks good, but a few things are missing:
1. There is no possibility to switch to satellite view, it only works in map view. This is important, because I should be able to see exactly where my house is.2. The location picker always comes up zoomed out almost to see the entire planet. This is not a huge problem, but it would be more user friendly if it remembered the last location and last zoom level that was used. Usually you want to add several geo-fences with different radius to the same center point so it doesn't make sense to search for your address every time.
Also, I noticed that coordinates cannot be entered manually anymore with latitude/longitude values, as before. This should not be a problem if the location picker allows you to pick any point on the map and does not snap to streets and known addresses. I live in a village and the location pickers used in other software usually snap to the main roads, which is not good for me because my house is several hundred meters away from the roads. At first glance it seems that in the Domoticz app the pin stays where you put it but I'll have to test when I get home to see if it really registers the picked point or in the background it actually saves a point which is snapped to a road.
Thanks,Andras
